crates/serde/src/optional.rs

```diff
@@ -0,0 +1,21 @@
+//! Serde functions for encoding optional values.
+//!
+//! This is defined as a "hex encoded unsigned integer", with a special case of 0 being `0x0`.
+//!
+//! A regex for this format is: `^0x([1-9a-f]+[0-9a-f]*|0)$`.
+//!
+//! This is only valid for human-readable [`serde`] implementations.
+//! For non-human-readable implementations, the format is unspecified.
+//! Currently, it uses a fixed-width big-endian byte-array.
+use serde::{Deserialize, Deserializer};
+
+/// For use with serde's `deserialize_with` on a sequence that must be
+/// deserialized as a single but optional (i.e. possibly `null`) value.
+pub fn or_default<'de, T, D>(deserializer: D) -> Result<T, D::Error>
```

Comment:
this only interprets "null" as default, right?

so perhaps we rename this to `null_as_default` or smth?

Reply:
Addressed in https://github.com/alloy-rs/alloy/pull/955/commits/f33ce645b4a08876ea65393e419a9f1b417e639e